substrate/frame/staking/src/lib.rs

```diff
@@ -1223,3 +1223,62 @@ impl BenchmarkingConfig for TestBenchmarkingConfig {
 	type MaxValidators = frame_support::traits::ConstU32<100>;
 	type MaxNominators = frame_support::traits::ConstU32<100>;
 }
+
+/// A result from a disabling decision made by [`DisablingStrategy`].
+pub struct DisablingDecision {
+	/// List of validator indices that should be disabled.
+	disable_offenders: Vec<u32>,
+}
+
+/// Input data for [`make_disabling_decision`]. Provides information about the offence so that the
```

Comment:
In this particular case I preferred struct because there are quite a few parameters and they are easy to misplace.
E.g. if `offender_idx`, `slash_era` and `era_now` are passed as a function parameter one might easily pass `slash_era` as `era_now` (or even offender_idx) and vice versa.
Packing the parameters as a struct makes this a bit harder to mess up - due to struct decomposition syntax you explicitly state which is which.

And last but not least - if another parameter needs to be added to `make_disabling_decision` in the future, extending the struct will be a bit easier compared to modifying the function signature.

But bottom line you are more involved in this codebase than me. I'll use separate parameters if you prefer to.